### PR TITLE
policy: Fix `cilium policy trace` output when only deny rules are applied

### PR DESCRIPTION
**Pull request by chez-shanpu**
Currently, deny rules are not counted in `*Repository.GetRulesMatching()`.
Therefore, this function returns false even though a rule is applied if it is a deny rule.
This causes a wrong output of `policy trace` commnad in some cases.

This PR makes this function to count ingress/egress deny rules.
Also, update the e2e test case for checking `policy trace` works correctly when there is only a deny rule exists.

### CODE REVIEW

File: pkg/policy/repository.go [hunk 1]
```diff
@@ -536,9 +536,15 @@ func (p *Repository) GetRulesMatching(lbls labels.LabelArray) (ingressMatch bool
...
 			}
+			if len(r.IngressDeny) > 0 {
+				ingressMatch = true
+			}
 			if len(r.Egress) > 0 {
 				egressMatch = true
 			}
+			if len(r.EgressDeny) > 0 {
+				egressMatch = true
+			}
```
Comment (pchaigno) [inline pkg/policy/repository.go:547]: Could we extend the unit tests in `pkg/policy/repository_deny_test.go` to cover this?

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -293,6 +293,11 @@ var _ = SkipDescribeIf(func() bool {
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
 			res.ExpectFail("%q can curl to %q", appPods[helpers.App3], clusterIP)
 
+			_, err = kubectl.CiliumPolicyAction(
+				namespaceForTest, l3Policy,
+				helpers.KubectlDelete, helpers.HelperTimeout)
+			Expect(err).Should(BeNil(), "Cannot delete L3 Policy")
```
Comment (pchaigno) [inline test/k8sT/Policies.go]: I'm not sure this change is correct. If we don't have a L3 allow policy for the L3 deny test, then everything is denied and we're not really testing the deny policy. /cc @aanm 

(FYI, if we keep this, it should be in a separate commit as AFAIU, it's independent from the change above.)

**Review by pchaigno — COMMENTED** [review]
Nice find!

File: test/k8sT/Policies.go (hunk 2 above)
Comment (aanm) [reply test/k8sT/Policies.go]: Correct. We shouldn't remove the L3 allow policy. Well, it might make sense to test without a L3 allow policy and with a L3 allow policy. On both cases the traffic should be dropped.

**Review by aanm — APPROVED** [review]
LGTM if Paul's feedback is addressed

File: test/k8sT/Policies.go (hunk 2 above)
Comment (chez-shanpu) [reply test/k8sT/Policies.go]: I see. I'll fix it :)

**Review by christarazi — APPROVED** [review]
LGTM pending Paul's comments, nice find! Could you add the context in the PR description to the commit msg itself also?

**Review by nebril — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]
:ship:

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
Both team review requests are covered and all required tests are passing. Marking ready to merge.

**Review by jrajahalme — APPROVED** [review]

File: pkg/policy/repository_deny_test.go [hunk 3]
```diff
@@ -260,6 +260,77 @@ func (ds *PolicyTestSuite) TestComputePolicyDenyEnforcementAndRules(c *C) {
...
+		Labels: tag,
+	}
+
+	// When no policy is applied.
+	ingressMatch, egressMatch := repo.GetRulesMatching(labels.LabelArray{bar, foo})
+	c.Assert(ingressMatch, Equals, false)
+	c.Assert(egressMatch, Equals, false)
+	ingressMatch, egressMatch = repo.GetRulesMatching(labels.LabelArray{bar, foo})
+	c.Assert(ingressMatch, Equals, false)
+	c.Assert(egressMatch, Equals, false)
```
Comment (chez-shanpu) [inline pkg/policy/repository_deny_test.go]: Duplicate lines

**Review by vadorovsky — APPROVED** [review]
